NMS-13885: Tuning broker docs (#4075)
OpenNMS/opennms: docs/modules/reference/pages/configuration/tuning-activemq.adoc

== Monitoring the ActiveMQ broker using the Karaf shell

> The `opennms:activemq-stats` command available via the Karaf shell can be used to show statistics about the embedded broker:

Reviewer: ```suggestion
Use the `opennms:activemq-stats` command available via the Karaf shell to show statistics about the embedded broker:
```

Users associated with the `MINION` role are restricted in such a way that prevents them from making RPC requests to other locations, but can otherwise read or write to the queues they need.

See the `authorizationPlugin` section in `$\{OPENNMS_HOME}/etc/opennms-activemq.xml` for details.

== Multi-tenancy with {page-component-title} and ActiveMQ

> The queue names used by {page-component-title} are prefixed with a constant value.

Reviewer: ```suggestion
The queue names {page-component-title} uses are prefixed with a constant value.
```

> If many {page-component-title} are configured to use the same broker, then these queues would end up being shared amongst the instances, which is not desired.

Reviewer: ```suggestion
If many {page-component-title} are configured to use the same broker, then these queues would end up being shared among the instances, which is not desired.
```

In order to isolate multiple instances on the same broker, you can customize the prefix by setting the value of the `org.opennms.instance.id` system property to something that is unique per instance.

.`$\{OPENNMS_HOME}/etc/opennms.properties.d/instance-id.properties`
[source, properties]
----
org.opennms.instance.id=MyNMS
----

Update the Minion's instance ID accordingly to match the {page-component-title} instance.

.`$\{MINION_HOME}/etc/custom.system.properties`.
[source, properties]
----
org.opennms.instance.id=MyNMS
----

CAUTION: If you change the instance ID setting when using the embedded broker, you will need to update the authorization section in the broker's configuration to reflect the updated prefix.

> The configuration can be modified with the file `$\{OPENNMS_HOME}/etc/opennms-activemq.xml`:

Reviewer: ```suggestion
Modify the configuration with the file `$\{OPENNMS_HOME}/etc/opennms-activemq.xml`:
```

  * Values should typically be under 100ms but may vary based on network latency
4. Resource graphs for the `JMX-Minion` service are populated and reasonable values

To interactively test RPC communication with a remote location use the `opennms:poll` command from the Karaf shell:

[source, karaf]
----
opennms:poll -l LOCATION -c org.opennms.netmgt.poller.monitors.IcmpMonitor 127.0.0.1 <1>
----

> <1> Replace `LOCATION` in the command above with the name of the Monitoring Location you want to test.

Reviewer: ```suggestion
<1> Replace `LOCATION` in the command above with the name of the monitoring location you want to test.
```
